### PR TITLE
Add integrations fix n+1

### PR DESCRIPTION
**Pull request by vflorin1995**
### Changes:

- Add integration tests for all 4 classes
- Optimize the n+1 query

file_1_before
file_1_after
file_2_before
file_2_after

### CODE REVIEW

File: spec/views/users/show.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      subject.last_3_posts.each do |ppost|
+        expect(page).to have_content(ppost.Title)
+        expect(page).to have_content(ppost.Text)
+      end
+
+      click_link('Hello1')
+      sleep(5)
+      expect(page).to have_current_path(user_post_path(subject, post))
+
+      visit user_path(subject)
```
Comment (mikethreels) [inline spec/views/users/show.html.erb_spec.rb]: - [ ] please put an `it` block before each test with a description of what the particular test is testing.

File: spec/views/users/index.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      expect(page).to have_content('test')
+      expect(page).to have_xpath("//img[@src='http://google.com']")
+      expect(page).to have_content('Number of posts: 0')
+
+      click_link('test')
+      sleep(5)
+      expect(page).to have_current_path(user_path(subject))
+    end
+  end
+end
```
Comment (mikethreels) [inline spec/views/users/index.html.erb_spec.rb]: - [ ] please put an `it` block before each test with a description of what the particular test is testing.

File: spec/views/posts/show.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      expect(page).to have_content(post.Title)
+      expect(page).to have_content(subject.Name)
+      expect(page).to have_content('Comments: 6')
+      expect(page).to have_content('Likes: 0')
+      expect(page).to have_content('This is my first post')
+      post.comments.each do |comment|
+        expect(page).to have_content(comment.Text)
+        expect(page).to have_content(comment.user.Name)
+      end
+    end
```
Comment (mikethreels) [inline spec/views/posts/show.html.erb_spec.rb:30]: - [ ] please put an `it` block before each test with a description of what the particular test is testing.

File: spec/views/posts/index.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      expect(page).to have_content('Number of posts: 4')
+      expect(page).to have_content('Hello20')
+      subject.posts.each do |ppost|
+        expect(page).to have_content(ppost.Title)
+        expect(page).to have_content(ppost.Text)
+      end
+      expect(page).to have_content('Hello3')
+      expect(page).to have_content('Comments: 6')
+      expect(page).to have_content('Likes: 0')
+
```
Comment (mikethreels) [inline spec/views/posts/index.html.erb_spec.rb:33]: - [ ] please put an `it` block before each test with a description of what the particular test is testing.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Florin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @vflorin1995, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Good job with test coverage
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**